jstreegrid.js

```diff
@@ -812,16 +829,76 @@
 				_this.redraw_node(rootNode, true);
 			});
 
-		};
+    };
+
+    this._domManipulation = null; // We'll store the column nodes in this object and an id for the grid-node that started the manipulation { id: "id of the node that started the manipulation", columns: { Key-Value-Pair col-No: Column }}
+
+    function _guid() {
+      function s4() {
+        return Math.floor((1 + Math.random()) * 0x10000)
+          .toString(16)
+          .substring(1);
+      }
+      return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
+        s4() + '-' + s4() + s4() + s4();
+    }
+    /*
+     * Trys to detach the tree columns on massive dom manipulations
+     */
+    this._detachColumns = function (id) {
+      // if the columns are not detached, then detach them
+      if (this._domManipulation == null) {
+        var cols = this._gridSettings.columns || [], treecol = this._gridSettings.treecol, mw = this.midWrapper;
+        this._domManipulation = { id: id, columns: {} };
+        for (var i = 0, len = cols.length; i < len; i++) {
+          if (treecol === i) {
+            continue;
+          }
+          this._domManipulation.columns[i] = mw.children(".jstree-grid-column-" + i)[0];
+          this._domManipulation.columns[i].parentNode.removeChild(this._domManipulation.columns[i]);
+        }
+      }
+      return this._domManipulation;
+    }
+
+    this._reattachColumns = function (id) {
+      if (this._domManipulation == null) { return false; }
+      if (this._domManipulation.id === id) {
+        var cols = this._gridSettings.columns || [], treecol = this._gridSettings.treecol, mw = this.midWrapper;
+        for (var i = 0, len = cols.length; i < len; i++) {
+          if (treecol === i) {
+            continue;
+          }
+          mw[0].appendChild(this._domManipulation.columns[i]);
+        }
+        this._domManipulation = null;
+      }
+      return true;
+    }
+
+    /*
+		 * Override open_node to detach the columns before redrawing child-nodes, and do reattach them afterwarts
+		 */
+    this.open_node = function (obj, callback, animation) {
+      var id = $.isArray(obj) ? _guid() : this.get_node(obj).id;
+      this._detachColumns(id);
+      var ret = parent.open_node.call(this,arguments);
+      this._reattachColumns(id);
+      return ret;
+    }
+
 		/*
 		 * Override redraw_node to correctly insert the grid
 		 */
-		this.redraw_node = function(obj, deep, is_callback, force_render) {
+    this.redraw_node = function (obj, deep, is_callback, force_render) {
+      var id = $.isArray(obj) ? _guid() : this.get_node(obj).id;
 			// first allow the parent to redraw the node
 			obj = parent.redraw_node.call(this, obj, deep, is_callback, force_render);
 			// next prepare the grid for a redrawn node - but only if ths node is not hidden (search does that)
-			if (obj && !this.is_hidden(obj)) {
-				this._prepare_grid(obj);
+      if (obj) {
```

Comment:
So this is the core of the rendering improvements? Detach, redo when not visual, reattach?